### DISCUSSION TITLE
Stress assisted diffusion coupling for anisotropic system

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#24366] [Q&A General]
Hello MOOSE experts,

I am trying to add the stress assisted coupling for the anisotropic case which includes the following term:

![image](https://github.com/idaholab/moose/assets/94848840/f4b344ab-13f0-4390-a7ff-b574ba2ea035)

In the above equation, diffusion (D) and expansion tensor \beta is 3 by 3 matrix.

I already did this coupling for the isotropic case; but, struggling to add for the anisotropic system.

Can you please point me in the right direction?

Thanks,
Avtar

**Comment by hugary1995** [comment]
1. Compute stress double contracted with beta.
2. Compute its gradient. I see that you are already aware of #22910
3. D is also a second order tensor, so the result is the mass flux (a vector).

**Reply by avtarsinghh1991** [reply]
Thanks @hugary1995 

I already calculated sigma_xx and sigma_yy and sigma_zz using `RankTwoCartesianComponent` in the material kernel and extracted them as material property using Auxkernel `MaterialRealAux`.

I am looking for the module to multiply them with expansion values and then add them to consider it a single variable. Then I can use the MatDiffusion kernel for the coupling. 

My another query is how to add prefactor in the MatDiffusion which can accommodate diffusivity as matrix.

Can you please me point me towards the modules which I can use?

Best,
Avtar

**Reply by GiudGiud** [reply]
Hello

Does the anisotropic diffusion work for you?
https://mooseframework.inl.gov/source/kernels/ADMatAnisoDiffusion.html

If not changing the diffusion coefficient in the kernel should not be too difficult

Guillaume

**Reply by avtarsinghh1991** [reply]
Hello @GiudGiud 

I need to add diffusivity as prefactor because there are other terms which needs to be included. 

For the isotropic case, I did it something like this:

```
[./constants]
    type = GenericConstantMaterial
    prop_names  = 'factor                 Omega   	D	R	           T'
    prop_values = '${pressure_coupling}  ${expansion_coeff}  ${diff}  ${gas_constant}	${temp}'
  [../]

  [./stress_coeff]
    type = ParsedMaterial
    function = 'factor*(D*Omega*c)/(R*T)'
    args = 'c'
    f_name = 'prefactor_pressure_coupling'
    material_property_names = 'factor D Omega	R	T'
  [../]
```

and then added it in the kernel as

```
 [stress_grad]
   type = MatDiffusion
   variable = 'c'
   v = 'pressure'
   diffusivity = 'prefactor_pressure_coupling'
 []
```

Is there a way to do same thing when diffusivity is matrix?
Then I can use MatAnisoDiffusion kernel for the coupling.

Other than that 
Can you also point me to the module which I can use to find the sum of `sigma_xx*beta_xx + sigma_yy*beta_yy + sigma_zz*beta_zz ` and then add it as material property for the coupling, the same way I did it for isotropic case?

Best,
Avtar

**Reply by GiudGiud** [reply]
See the object I pasted for a start on anisotropic diffusion

**Reply by GiudGiud** [reply]
Though I m sure there's anisotropic stress models in the tensor mechanics module you could look at as well

**Reply by hugary1995** [reply]
This will be quite a stretch without writing some C++.

**Reply by avtarsinghh1991** [reply]
@hugary1995  Ok. Can you please point me to the closest kernel (.C subroutine) so that I can build on top of it? Thanks.
